app/src/main/java/com/oracle/javafx/scenebuilder/app/menubar/MenuBarController.java

```diff
@@ -1171,6 +1176,24 @@ public String getTitle() {
         insertMenu.setOnMenuValidation(onCustomPartOfInsertMenuValidationHandler);
         
         windowMenu.setOnMenuValidation(onWindowMenuValidationHandler);
+
+        // Modify the Check Updates menu item if there is an update available.
+        // Icons by Font Awesome (https://fontawesome.com/license/free) under CC BY 4.0 License
+        AppSettings.getLatestVersion(latestVersion -> {
+            var updateAvailable = AppSettings.isUpdateAvailable();
+            Image icon = new Image(MenuBarController.class
+                    .getResource(updateAvailable ? "download_icon.png" : "check_icon.png")
+                    .toExternalForm());
+            ImageView iconView = new ImageView(icon);
+
+            Platform.runLater(() -> {
+                checkUpdatesMenuItem.setGraphic(iconView);
+                checkUpdatesMenuItem.disableProperty().setValue(!updateAvailable);
```

Comment:
In FXML, I'd disable the item by default and use `menu.title.check.updates.ok` text with `"check_icon.png"` graphic. If you open the .fxml in Scene Builder you can edit the update menu item.

Then the code just checks if the update is available and, if it is, overwrites it with "download_icon" and "menu.title.check.updates.available" text.

Reply:
Thanks for that snippet, my google-fu was failing me. Had to prefix the url with '@' though. 

Pushed changes for all these suggestions now. 👍